Row.c

```diff
@@ -190,94 +190,112 @@ RowField RowField_keyAt(const Settings* settings, int at) {
-      RichString_appendnAscii(str, largeNumberColor, buffer, len);
-      number %= 1000;
-      len = xSnprintf(buffer, sizeof(buffer), "%03lluG ", number);
-      RichString_appendnAscii(str, gigabytesColor, buffer, len);
-   } else if (number < 100ULL * ONE_G) {
-      //2 digit TB, 1 digit GB
-      len = xSnprintf(buffer, sizeof(buffer), "%2llu", number / ONE_G);
-      RichString_appendnAscii(str, largeNumberColor, buffer, len);
-      number = (number % ONE_G) * 10 / ONE_G;
-      len = xSnprintf(buffer, sizeof(buffer), ".%1llu", number);
-      RichString_appendnAscii(str, gigabytesColor, buffer, len);
-      RichString_appendAscii(str, largeNumberColor, "T ");
-   } else if (number < 10000ULL * ONE_G) {
-      //3 digit TB or 1 digit PB, 3 digit TB
-      number /= ONE_G;
-      len = xSnprintf(buffer, sizeof(buffer), "%4lluT ", number);
-      RichString_appendnAscii(str, largeNumberColor, buffer, len);
+   number = hundredths / 100;
+   hundredths %= 100;
+   if (number < 100) {
+      if (number < 10) {
+         // 1 digit + decimal point + 2 digits
+         // "9.76G", "9.99G", "9.76T", "9.99T", etc.
+         len = xSnprintf(buffer, sizeof(buffer), "%1u", (unsigned int)number);
+         RichString_appendnAscii(str, color, buffer, len);
+         len = xSnprintf(buffer, sizeof(buffer), ".%02u", (unsigned int)hundredths);
+      } else {
+         // 2 digits + decimal point + 1 digit
+         // "97.6M", "99.9M", "10.0G", "99.9G", etc.
+         len = xSnprintf(buffer, sizeof(buffer), "%2u", (unsigned int)number);
+         RichString_appendnAscii(str, color, buffer, len);
+         len = xSnprintf(buffer, sizeof(buffer), ".%1u", (unsigned int)hundredths / 10);
+      }
+      RichString_appendnAscii(str, prevUnitColor, buffer, len);
+      len = xSnprintf(buffer, sizeof(buffer), "%c ", unitPrefixes[i]);
+   } else if (number < 1000) {
+      // 3 digits
+      // "100M", "999M", "100G", "999G", etc.
+      len = xSnprintf(buffer, sizeof(buffer), "%4u%c ", (unsigned int)number, unitPrefixes[i]);
    } else {
-      //2 digit PB and above
-      len = xSnprintf(buffer, sizeof(buffer), "%4.1lfP ", (double)number / ONE_T);
-      RichString_appendnAscii(str, largeNumberColor, buffer, len);
+      // 1 digit + 3 digits
+      // "1000M", "9999M", "1000G", "9999G", etc.
+      assert(number < 10000);
+
+      len = xSnprintf(buffer, sizeof(buffer), "%1u", (unsigned int)number / 1000);
+      RichString_appendnAscii(str, nextUnitColor, buffer, len);
+      len = xSnprintf(buffer, sizeof(buffer), "%03u%c ", (unsigned int)number % 1000, unitPrefixes[i]);
    }
+   RichString_appendnAscii(str, color, buffer, len);
+   return;
+
+invalidNumber:
+   if (coloring)
+      color = CRT_colors[PROCESS_SHADOW];
+   RichString_appendAscii(str, color, "  N/A ");
+   return;
 }
 
-void Row_printKBytes(RichString* str, unsigned long long number, bool coloring) {
+void Row_printBytes(RichString* str, unsigned long long number, bool coloring) {
```

Comment:
Things below 1K don't really show as bytes with this despite the naming. We could extend the colors to use `PROCESS_SHADOW` for such small values though (only available if called through this wrapper) …

Reply:
I was keeping the original behavior of `Row_printBytes` while extending the function of `Row_printKBytes`. The original `Row_printBytes` prints 0 for everying below 1K (1024). My patch didn't change any of that behavior. It would be another PR for improving `Row_printBytes` after this...